Dockerfiles/agent/entrypoint.sh

```diff
@@ -8,35 +8,34 @@
 
 ##### Core config #####
 
-if [[ -z $DD_API_KEY ]]; then
-    echo "You must set an DD_API_KEY environment variable to run the Datadog Agent container"
+if [[ -z "${DD_API_KEY}" ]]; then
+    echo "You must set an DD_API_KEY environment variable to run the Datadog Agent container" >&2
     exit 1
 fi
 
-if [[ -z $DD_DD_URL ]]; then
+if [[ -z "${DD_DD_URL}" ]]; then
     export DD_DD_URL="https://app.datadoghq.com"
 fi
 
-if [[ -z $DD_DOGSTATSD_SOCKET ]]; then
+if [[ -z "${DD_DOGSTATSD_SOCKET}" ]]; then
     export DD_DOGSTATSD_NON_LOCAL_TRAFFIC=1
-else
-    if [[ -e $DD_DOGSTATSD_SOCKET ]]; then
-        if [[ -S $DD_DOGSTATSD_SOCKET ]]; then
-            echo "Deleting existing socket at ${DD_DOGSTATSD_SOCKET}"
-            rm $DD_DOGSTATSD_SOCKET
-        else
-            echo "${DD_DOGSTATSD_SOCKET} exists and is not a socket, please check your volume options"
-            exit 1
-        fi
+elif [[ -e "${DD_DOGSTATSD_SOCKET}" ]]; then
+    if [[ -S "${DD_DOGSTATSD_SOCKET}" ]]; then
+        echo "Deleting existing socket at ${DD_DOGSTATSD_SOCKET}"
+        rm -v "${DD_DOGSTATSD_SOCKET}" || exit $?
```

Comment:
Do we want to exit here ?

Reply:
Yes, that means dsd uds can't init, only known cause would be a read-only volume